ethers-etherscan/src/contract.rs

```diff
@@ -234,6 +234,14 @@ impl Client {
     /// # }
     /// ```
     pub async fn contract_abi(&self, address: Address) -> Result<Abi> {
+        // apply caching
+        if let Some(ref cache) = self.cache {
+            let path = cache.join("abis").join(format!("{:?}.json", address));
+            let reader = std::io::BufReader::new(std::fs::File::create(path)?);
+            let abi = serde_json::from_reader(reader)?;
+            return Ok(abi)
```

Comment:
should we instead continue with the request if this results in an error, like file does not exist?